client/extensions/zoninator/components/recent-posts-dropdown/index.jsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { find, flowRight, toArray } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryPosts from 'components/data/query-posts';
+import SelectDropdown from 'components/select-dropdown';
+import { getSitePostsForQuery } from 'state/posts/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+const recentPostsQuery = {
+	status: 'publish,future',
+	number: 10,
+};
+
+class RecentPostsDropdown extends PureComponent {
+	static propTypes = {
+		ignored: PropTypes.array,
+		onSelect: PropTypes.func.isRequired,
+		recentPosts: PropTypes.array,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+	}
+
+	static defaultProps = {
+		recentPosts: [],
+		ignored: [],
+	}
+
+	handleSelect = option => {
+		const { onSelect, recentPosts } = this.props;
+		const slug = option.value;
+
+		onSelect( find( recentPosts, { slug } ) );
+	}
+
+	render() {
+		const { ignored, recentPosts, siteId, translate } = this.props;
+
+		const className = 'zoninator__recent-posts-dropdown';
+
+		// getSitePostsForQuery can return null, hence the need to ensure recentPosts is an array.
+		const options = toArray( recentPosts )
```

Comment:
What about handling this in `connectComponent`? FWIW, in the theme showcase, I'm simpy falling back to `|| []` in `mapStateToProps`. (Not that that's terribly elegant.)